filebeat/templates/deployment.yaml

```diff
@@ -0,0 +1,141 @@
+# Deploy singleton instance in the whole cluster for some unique data sources, like aws input
+{{- if .Values.deployment.enabled }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: '{{ template "filebeat.fullname" . }}'
+  labels:
+    app: '{{ template "filebeat.fullname" . }}'
+    chart: '{{ .Chart.Name }}-{{ .Chart.Version }}'
+    heritage: '{{ .Release.Service }}'
+    release: '{{ .Release.Name }}'
+  {{- if .Values.deployment.annotations}}
```

Comment:
if we add annotations here, we should add them also in daemonset template for consistency